benchmark/test/test_metrics_parser.py

```diff
@@ -0,0 +1,342 @@
+# Copyright 2016 ClusterHQ Inc.  See LICENSE file for details.
+
+from benchmark.metrics_parser import (
+    mean, container_convergence, cpu_usage_for_process,
+    wallclock_for_operation, request_latency, handle_cputime_metric,
+    handle_wallclock_metric
+)
+from flocker.testtools import TestCase
+
+
+class MetricsParserTests(TestCase):
+    """
+    Tests for the metrics parsing script.
+    """
+
+    def test_mean_returns_floating_point_number(self):
+        values = [1, 2, 3, 4]
+        mean_result = mean(values)
+        self.assertIsInstance(mean_result, float)
+
+    def test_mean_returns_none_for_no_values(self):
+        values = []
+        self.assertEqual(mean(values), None)
+
+    def test_mean_correctly_calculates_mean(self):
+        values = [1, 2, 3, 4, 5, 6]
+        self.assertEqual(mean(values), 3.5)
+
+    def test_cpu_usage_for_process_no_matching_results(self):
+        """
+        cputime_for_process only considers results which have the
+        'cputime' metric type and match the specified process name.
+        None is returned if no results match.
+        """
+        process_name = 'test-process'
+        results = [
+            {
+                'metric': {
+                    'type': 'wallclock'
+                },
+                'process': process_name
+            },
+            {
+                'metric': {
+                    'type': 'cputime'
+                },
+                'process': 'another-process'
+            },
+        ]
+        cputime_result = cpu_usage_for_process(results, process_name)
+        self.assertEqual(cputime_result, None)
+
+    def test_cpu_usage_for_process_calculates_result(self):
+        """
+        cputime_for_process correctly calculates the mean CPU percentage
+        for a process by dividing the cputime by the wallclock time for
+        each sample
+        """
+        process_name = 'test-process'
+        results = [
+            {
+                'metric': {
+                    'type': 'cputime'
+                },
+                'process': process_name,
+                'value': 10,
+                'wallclock': 50
+            },
+            {
+                'metric': {
+                    'type': 'cputime'
+                },
+                'process': process_name,
+                'value': 30,
+                'wallclock': 100
+            },
+        ]
+        cputime_result = cpu_usage_for_process(results, process_name)
+        self.assertEqual(cputime_result, 0.25)
```

Comment:
:question: This seems like the wrong answer. The total CPU time is 40 seconds and the total walltime is 150 seconds. Hence the average is 26.67% CPU load.

If we observed 90% load for 1 second, but another sample showed 10% load for 60 seconds, we'd probably want the result to be closer to 10% than to 90%.

OTOH, if the above is true, why run 3 samples when we could just run 1 sample for 3 times as long? Given our sample denominators are likely to be much closer than this, it probably doesn't matter much in practice.